src/main/java/spoon/reflect/visitor/chain/CtQueryImpl.java

```diff
@@ -90,126 +66,122 @@ public CtQueryImpl() {
 	 * @param input
 	 * @return this to support fluent API
 	 */
-	public CtQuery<O> addInput(O input) {
+	public CtQueryImpl addInput(Object... input) {

```

Comment:
Because there are some interesting methods (evaluate, ...) which are available on CtQueryImpl only and if I cast query to CtQueryImpl once then I want to have CtQueryImpl after call of addInput too. It is correct implementation of fluent API